src/utils/url_utils.js

```diff
@@ -1,5 +1,29 @@
-export function parseUrl(str) {
-  const parser = global.document.createElement('a');
-  parser.href = str;
-  return parser;
+export function getLocationFromUrl(href) {
+  var match = href.match(
+    /^(https?:)\/\/(([^:/?#]*)(?::([0-9]+))?)([/]{0,1}[^?#]*)(\?[^#]*|)(#.*|)$/
+  );
+  return (
+    match && {
+      href: href,
+      protocol: match[1],
+      host: match[2],
+      hostname: match[3],
+      port: match[4],
+      pathname: match[5],
+      search: match[6],
+      hash: match[7]
+    }
+  );
+}
+
+export function getOriginFromUrl(url) {
+  if (!url) {
+    return undefined;
+  }
+  var parsed = getLocationFromUrl(url);
+  var origin = parsed.protocol + '//' + parsed.hostname;
```

Comment:
... this is also a source of an error if you just pass in a string:

```
> getLocationFromUrl('banana')
< null
```

If there is no match then `null` is returned.

Reply:
good catch. pushed a fix.